index.js

```diff
@@ -288,11 +317,19 @@ function Hook (modules, options, onrequire) {
 
 Hook.prototype.unhook = function () {
   this._unhooked = true
+
   if (this._require === Module.prototype.require) {
     Module.prototype.require = this._origRequire
-    debug('unhook successful')
+    debug('require unhook successful')
+  } else {
+    debug('require unhook unsuccessful')
+  }
+
+  if (this._getBuiltinModule === process.getBuiltinModule) {
```

Comment:
Can we also guard this on `typeof process.getBuiltinModule === 'function'` or `!== undefined`? Currently `undefined === undefined` and the debug logging is misleading on earlier versions of Node.js:

```
% cat foo.js
console.log('Running node', process.version)
var {Hook} = require('./');
var hook = new Hook(['http'], (mod, name, baseDir) => {
    console.log('Hooked %j', name);
});
require('http');
hook.unhook();

% DEBUG=require-in-the-middle node foo.js
Running node v18.20.4
  require-in-the-middle registering require hook +0ms
  require-in-the-middle processing core module require('http'): http +0ms
  require-in-the-middle calling require hook: http +1ms
Hooked "http"
  require-in-the-middle returning module: http +0ms
  require-in-the-middle require unhook successful +0ms
  require-in-the-middle process.getBuiltinModule unhook successful +0ms
```

Reply:
Good catch! Thanks for such a thorough review 💯 

Fixed now in 93cf4ec 👍